libs/num/numeric.go

```diff
@@ -0,0 +1,83 @@
+package num
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+type Numeric struct {
+	AsUint    *Uint
+	AsDecimal *Decimal
+}
+
+func (n *Numeric) Clone() *Numeric {
+	nn := &Numeric{}
+
+	if n.AsUint != nil {
+		nn.AsUint = n.AsUint
```

Comment:
The uint should be cloned

Reply:
Changed this to use the same address for the Uint and a pointer to a copy for decimal -> https://github.com/vegaprotocol/vega/pull/6935/commits/db7f4cd7157e7edeeb1c9c464b9094521d5be6fb#diff-01e84bbc9c49dbdd4b9e45ec724423ea40a7c808577eebb58ed4f69f25ac4755R14-R27
Thanks!
* There are a lot of places in the code where this is not done in the deep clones, should we track and fix them?